graph/src/blockchain/block_stream.rs

```diff
@@ -403,13 +412,28 @@ pub trait SubstreamsMapper<C: Blockchain>: Send + Sync {
                 };
 
                 let block = self.decode_triggers(&logger, &clock, &map_output).await?;
+                log_data.last_seen_block = block.block.number() as u64;
 
                 Ok(Some(BlockStreamEvent::ProcessBlock(
                     block,
                     FirehoseCursor::from(cursor.clone()),
                 )))
             }
 
+            Some(SubstreamsMessage::Progress(progress)) => {
+                if log_data.last_progress.elapsed() > Duration::from_secs(30) {
+                    info!(&logger, "{}", log_data.info_string(&progress));
+                    debug!(&logger, "{}", log_data.debug_string(&progress));
```

Comment:
This means you get all 3 logs if debug level is tracing, maybe it would be worth making sure you'd only get the appropriate information once?